bin/manage.py

```diff
@@ -0,0 +1,406 @@
+    except Exception as e:
+        log.error(e, 'unable to get primary status')
+        sys.exit(1)
+
+    if is_mongo_primary:
+        mongo_update_replset_config(local_mongo, hostname)
+
+# ---------------------------------------------------------
+
+def is_mongo_up(local_mongo, timeout=MONGO_RETRY_TIMES):
+    '''
+    check to see if mongo is up yet, retying the given number of times
+    '''
+    
+    while True:
+        timeout -= 1
+        if timeout <= 0:
+            log.errors('unable to connect to mongodb after %i times' % MONGO_RETRY_TIMES)
+            return False
+
+        try:
+            # check that mongo is up
+            server_info = local_mongo.server_info()
+            if not server_info['ok']:
+                log.info('Mongo response not "ok" on %s; retrying...' % ip)
+                sleep(1)
+                continue
+            break
+        except (AutoReconnect, ServerSelectionTimeoutError) as e:
+            log.info('Mongo not yet available on %s; retrying...' % ip)
+            sleep(1)
+            continue
+        except (ConnectionFailure, NetworkTimeout, NotMasterError) as e:
+            # TODO retry like AutoReconnect error above?
+            log.info(e)
+            return False
+        except Exception as e:
+            # just bail on unexpected exceptions when trying to connect
+            log.error(e)
+            return False
+    return True
+
+@debug
+def mongo_update_replset_config(local_mongo, hostname):
+    '''
+    called from the primary node to update the replica config in mongo
+    using the current set of healthy mongo containers listed in consul
+    '''
+    try:
+        repl_config = local_mongo.admin.command('replSetGetConfig')
+        if not repl_config['ok']:
+            raise Exception('could not get replica config: %s' % repl_config['errmsg'])
+        repl_config = repl_config['config']
+        # TODO use consul.agent.health() instead?
+        consul_services = consul.agent.services()
+
+        mongos_in_consul = []
+        for x in consul_services:
+            host = consul_to_mongo_hostname(consul_services[x]['ID'])
+            if host:
+                mongos_in_consul.append(host)
+        if not len(mongos_in_consul):
+            return
+        if not hostname + ':27017' in mongos_in_consul:
+            return
+
+        max_id = 0
+        changed = False
+        new_members = []
+        for member in repl_config['members']:
+            if member['_id'] > max_id:
+                max_id = member['_id']
+            if member['host'] in mongos_in_consul:
+                mongos_in_consul.remove(member['host'])
+                new_members.append(member)
+            else:
+                changed = True
+        for host in mongos_in_consul:
+            changed = True
+            max_id += 1
+            new_members.append({
+                    '_id': max_id,
+                    'host': host,
+                })
```

Comment:
That code does look a little clearer; I'll incorporate it in.

As for the `_id` on each member, from my theoretical perspective, as long as the old `_id` is gone mongodb won't care if a new one is added with the same `_id`.  Also, from a practical perspective this is true because I just tested scaling from 3 nodes to 2 and back to 3.  The new 3rd mongo member was a completely new container and it took the same `_id` in the mongo config.